File: src/main/java/com/hellsfood/api/users/data/Role.java
package com.hellsfood.api.users.data;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;

import org.hibernate.annotations.DynamicInsert;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@DynamicInsert
@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Role {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(length = 32, nullable = false, unique = true)
	private String roleName;

	@ManyToMany(mappedBy = "roles")
	@JsonIgnore
	private List<User> users;

	public Role(String roleName) {
		this.roleName = roleName;
	}
}

File: src/main/kotlin/com/hellsfood/api/users/data/Role.kt
package com.hellsfood.api.users.data

import com.fasterxml.jackson.annotation.JsonIgnore
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.ToString
import org.hibernate.annotations.DynamicInsert
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.ManyToMany

@DynamicInsert
@Entity
@Getter
@NoArgsConstructor
@AllArgsConstructor
@ToString
class Role {
    @Id
    private val id: Long? = null
    private val roleName: String? = null

    @ManyToMany(mappedBy = "roles")
    @JsonIgnore
    @ToString.Exclude
    private val users: List<User>? = null
}